2.7/s2i/bin/assemble

```diff
@@ -18,23 +18,31 @@ function virtualenv_bin() {
   fi
 }
 
-# Install pipenv to the separate virtualenv to isolate it
+# Install pipenv or micropipenv to the separate virtualenv to isolate it
 # from system Python packages and packages in the main
 # virtualenv. Executable is simlinked into ~/.local/bin
 # to be accessible. This approach is inspired by pipsi
 # (pip script installer).
-function install_pipenv() {
-  echo "---> Installing pipenv packaging tool ..."
+function install_tool() {
+  echo "---> Installing $1 packaging tool ..."
   pip install -U virtualenv
-  VENV_DIR=$HOME/.local/venvs/pipenv
+  VENV_DIR=$HOME/.local/venvs/$1
   virtualenv_bin "$VENV_DIR"
-  $VENV_DIR/bin/pip --isolated install -U pipenv
+  $VENV_DIR/bin/pip --isolated install -U $1$2  # Combines package name with [extras] if [extras] is defined as $2
   mkdir -p $HOME/.local/bin
-  ln -s $VENV_DIR/bin/pipenv $HOME/.local/bin/pipenv
+  ln -s $VENV_DIR/bin/$1 $HOME/.local/bin/$1
 }
 
 set -e
 
+# First of all, check that we don't have disallowed combination of ENVs
+if [[ ! -z "$ENABLE_PIPENV" && ! -z "$ENABLE_MICROPIPENV" ]]; then
+  echo "ERROR: Pipenv and micropipenv cannot be enabled at the same time!"
+  # podman/buildah does not relay this exit code but it will be fixed hopefuly
+  # https://github.com/containers/buildah/issues/2305
+  exit 3
```

Comment:
Do we really want exit 3? I guess, enough is exit 1.

Reply:
The plan here is that when [this change in buildah](https://github.com/containers/buildah/pull/2306) will be released, we will be able to do a better check for the specific exit code which is not possible now.
In other words, I am not able to check that the unsupported combination of micropipenv and pipenv is the issue why the build failed but with a specific exit code and a newer buildah, I'll be able to do it in the future.
WDYT?

Reply:
Sounds good! Maybe I would choose a higher code so there's less chance of something else throwing code 3, but that's probably too paranoid :)